workers/ohsome_quality_analyst/indicators/mapping_saturation/indicator.py

```diff
@@ -35,162 +36,177 @@ def __init__(
-        # saturation will be calculated
-        sigmoid_curve = sigmoidCurve()
-        # TODO: Change the dict to three parameters
-        ydata_for_sat = sigmoid_curve.getBestFittingCurve(
-            {
-                "timestamps": self.timestamps,
-                "results": self.values,
-                "results_normalized": self.values_normalized,
-            }
+        self.best_fit = min(self.fitted_models, key=lambda m: m.mae)
+        logging.info("Best fitting model: " + self.best_fit.name)
+        # Calculate saturation: The slope of the last 3 years of the fitted curve
+        self.saturation = (
+            np.interp(xdata[-36], xdata, self.best_fit.fitted_values)
+        ) / (np.interp(xdata[-1], xdata, self.best_fit.fitted_values))
+        self.result.value = self.values[-1] / self.best_fit.asymptote  # Completeness
+        description = Template(self.metadata.result_description).substitute(
+            saturation=self.saturation,
+            completeness=self.result.value,
         )
-        if ydata_for_sat[0] != "empty":
-            # check if data are more than start stadium
-            # The end of the start stage is defined with
-            # the maximum of the curvature function f''(x)
-            # here: simple check <= 2
-            # TODO implement function from MA for start stadium
-            #
-            # For buildings-count in a small area, this could return a wrong
-            # interpretation, eg a little collection of farm house and buildings
-            # with eg less than 8 buildings, but all buildings are mapped, the value
-            # would be red, but its all mapped...
-            # calculate/define traffic light value and label
-            if max(self.values) <= 2:
-                # start stadium, some data are there, but not much
-                self.saturation = 0
-            else:
-                indices = list(range(len(self.values)))
-                # calculate slope/growth of last 3 years
-                # take value in -36. month and value in last month of data
-                early_x = indices[-36]
-                last_x = indices[-1]
-                # get saturation level within last 3 years
-                self.saturation = sigmoid_curve.getSaturationInLast3Years(
-                    early_x, last_x, indices, ydata_for_sat
-                )
-                # if earlyX and lastX return same y value
-                # (means no growth any more), then
-                # getSaturationInLast3Years returns 1.0
-
-            # if saturation == 1.0:
-            #    growth should be 0.0
-            self.growth = 1 - self.saturation
-
-            description = Template(self.metadata.result_description).substitute(
-                saturation=self.saturation,
-                growth=self.growth,
+        if self.result.value >= self.upper_threshold:
+            self.result.label = "green"
+            self.result.description = (
```

Comment:
Good catch. I fixed this by relying only on the result description and removing the description for each label. I think it is not necessary to provide a different description/interpretation for each label since it is a simple ratio in %. What do you think?

E.g. the description now is simply: Thea rea of interest is xx % complete.